src/shared/components/results/RenderColumnsInCard.tsx

```diff
@@ -0,0 +1,47 @@
+import { memo, ReactNode } from 'react';
+
+import { ColumRenderer } from '../../types/columnConfiguration';
+
+import styles from './styles/render-columns-in-card.module.scss';
+
+/**
+ * Renders a column renderer or a list of renderers for usage within a card with
+ * with all the corresponding wrapping markup and classes
+ */
+const RenderColumnsInCard = <Schema extends Record<string, unknown>>({
+  renderers,
+  data,
+}: {
+  /** Renderer or list of renderers to render with the data */
+  renderers:
+    | Array<ColumRenderer<Schema> | undefined>
+    | ColumRenderer<Schema>
+    | undefined;
+  /** Data to pass to the renderer(s) */
+  data: Schema;
+}) => {
+  const listOfRenderers = Array.isArray(renderers) ? renderers : [renderers];
+  const renderedNodes: ReactNode[] = listOfRenderers
+    .map((renderer, index) => {
+      const rendered = renderer?.render(data);
+      if (!rendered || !renderer) {
+        return null;
+      }
+      return (
+        <span
+          className={styles['result-card__info-bit']}
+          key={typeof renderer.label === 'string' ? renderer.label : index}
+        >
+          <strong>{renderer.label}: </strong>
+          {rendered}
+        </span>
+      );
+    })
+    .filter(Boolean);
+
+  return renderedNodes.length ? (
+    <div className={styles['result-card__info-container']}>{renderedNodes}</div>
+  ) : null;
+};
+
+export default memo(RenderColumnsInCard) as typeof RenderColumnsInCard;
```

Comment:
In this instance `data` is deeply nested and `renderers` is an array of functions so shallow comparison between previous props and next props won't work and instead reference comparison is used and I don't think the props references will be stable. Providing our own `areEqual` and just looking at the accession/id could be a good comparison. I played around with the facets on proteomes results view to see when it does re-render:

```
const areEqual = (p, n) => {
  console.log(p.data.id, n.data.id, p == n, p.data.id == n.data.id);
  // return p == n;
  return p.data.id == n.data.id;
};
export default memo(
  RenderColumnsInCard,
  areEqual
) as typeof RenderColumnsInCard;

```
One issue in my suggestion is that we won't always have `id` available so we would have to somehow indicate what to compare on or just try id-like attributes one by one (try id and if that's not there try accession and if that's not there try...).

Reply:
About the id, on top of the limitation that there might not always be an `id` field so we would need to have more complexe logic, I'm not a fan of trying to check if the id is the same anyway, as differents requests (with the difference in the fields for example) will return different objects with the same id so we wouldn't detect new objects when we actually do need to.
Also, I'm assuming in the case where our data object is meant to stay the same, it will, as it is the raw object returned by the API through `useDataAPI` and `useDataAPI` is in charge of returning the same actual object at every render.

Regarding the `renderers` props, since this selection of renderers is fixed for any given card, I always define that array outside of the component in order for it to be always the same array.

Reply:
All good points. When should we be preventing a re-render - when will `memo` stop react from re-rendering? Facets will trigger a re-render as it's coming from a different query.

Reply:
I don't believe the logic within these components is so complex that it would be worth having an extra complex logic to determine if we should re-render or not.
When facets changes, the data object will be new indeed, but likely the resulting list of entries to display would change too anyway so it would be OK.
On the top of my mind the only case where this might be useful is when checking/unchecking the selected checkboxes, that by nature forces the rerendering of all the cards, while most all of their content is unchanged (apart from one specific checkbox). Ideally we'd want the whole rest of the card to be a memoized component and the checkbox to be separated, but that falls a bit further out of the scope of this PR I believe.

Reply:
Good point about the checkboxes